web/src/main/java/se/inera/intyg/webcert/web/service/facade/impl/certificatefunctions/SendCertificateFunctionImpl.java

```diff
@@ -70,55 +73,75 @@ public class SendCertificateFunctionImpl implements SendCertificateFunction {
     private final List<String> allowedCertificateTypes = List.of(LuaenaEntryPoint.MODULE_ID, LisjpEntryPoint.MODULE_ID,
-        TsBasEntryPoint.MODULE_ID, LuseEntryPoint.MODULE_ID, LuaefsEntryPoint.MODULE_ID);
+        TsBasEntryPoint.MODULE_ID, LuseEntryPoint.MODULE_ID, LuaefsEntryPoint.MODULE_ID, TsDiabetesEntryPoint.MODULE_ID);
     private static final String SEND_TO_FK = "Skicka till Försäkringskassan";
     private static final String SEND_TO_TS = "Skicka till Transportstyrelsen";
 
     @Override
     public Optional<ResourceLinkDTO> get(Certificate certificate) {
         if (!allowedCertificateTypes.contains(certificate.getMetadata().getType()) || isSent(certificate) || isReplacementSigned(
-            certificate)) {
+            certificate) || isWrongMajorVersion(certificate)) {
             return Optional.empty();
         }
 
-        if (certificate.getMetadata().getType().equals(TsBasEntryPoint.MODULE_ID) && !certificate.getMetadata().isLatestMajorVersion()) {
-            return Optional.empty();
-        }
+        return getResourceLinkDTO(certificate).get(certificate.getMetadata().getType());
+    }
 
-        if (shouldSendToFk(certificate)) {
-            return Optional.of(
-                ResourceLinkDTO.create(
-                    ResourceLinkTypeDTO.SEND_CERTIFICATE,
-                    SEND_TO_FK,
-                    SEND_TO_FK_DESCRIPTION,
-                    SEND_BODY_LUAENA,
-                    true));
-        } else if (certificate.getMetadata().getType().equalsIgnoreCase(LisjpEntryPoint.MODULE_ID)) {
-            return Optional.of(
-                ResourceLinkDTO.create(
-                    ResourceLinkTypeDTO.SEND_CERTIFICATE,
-                    SEND_TO_FK,
-                    SEND_TO_FK_DESCRIPTION,
-                    hasShortSickleavePeriod(certificate) ? SEND_BODY_SHORT_SICKLEAVE_PERIOD : SEND_BODY_LISJP,
-                    true
-                )
-            );
-        } else {
-            return Optional.of(
-                ResourceLinkDTO.create(
-                    ResourceLinkTypeDTO.SEND_CERTIFICATE,
-                    SEND_TO_TS,
-                    SEND_TO_TS_DESCRIPTION,
-                    SEND_BODY_TS_BAS,
-                    true
-                )
-            );
-        }
+    private static boolean isWrongMajorVersion(Certificate certificate) {
+        return (certificate.getMetadata().getType().equals(TsBasEntryPoint.MODULE_ID)
+            || certificate.getMetadata().getType().equals(TsDiabetesEntryPoint.MODULE_ID)) && !certificate.getMetadata()
+            .isLatestMajorVersion();
+    }
+
+    private Map<String, Optional<ResourceLinkDTO>> getResourceLinkDTO(Certificate certificate) {
+        return Map.of(
+            LisjpEntryPoint.MODULE_ID, getResourceLinkDTOFkLisjp(certificate),
+            LuseEntryPoint.MODULE_ID, getResourceLinkDTOFk(),
```

Comment:
Suggesting a switch-case here instead. Seems that only one resource link is fetched on each call, no need to prepare all of them each time(?)

Reply:
Thats a great suggestion, will do!